src/utils/convert.hpp

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef CONVERT_HPP_
+#define CONVERT_HPP_
+
+#include <aos/common/cloudprotocol/alerts.hpp>
+#include <aos/common/cloudprotocol/envvars.hpp>
+#include <aos/common/cloudprotocol/log.hpp>
+#include <aos/common/monitoring/monitoring.hpp>
+#include <aos/common/types.hpp>
+#include <aos/sm/networkmanager.hpp>
+
+#include <servicemanager/v4/servicemanager.grpc.pb.h>
+
+namespace aos::sm::utils {
+
+/**
+ * Converts byte array to aos.
+ *
+ * @param arr array to convert.
+ * @return const Array<uint8_t>.
+ */
+const Array<uint8_t> ConvertByteArrayToAos(const std::string& arr);
+
+/**
+ * Converts aos error to protobuf error.
+ *
+ * @param error aos error.
+ * @return iamanager::v5::ErrorInfo.
+ */
+::common::v1::ErrorInfo ConvertAosErrorToProto(const Error& error);
```

Comment:
It looks like, all conversion between Aos and PB  should be in aos_core_common_cpp but in different files depends on PB headers available for specific aos service.

Reply:
Sounds reasonable.
But I'd wait Mykola Solianko's PR with iam client to be merged first (to reduce possible merge  conflicts)